src/negate.ts

```diff
@@ -0,0 +1,23 @@
+import type Include from "./types/Include";
+
+/**
+ * Create a function that returns the opposite result of a predicate function
+ *
+ * @example
+ * ```ts
+ * const isDefined = negate(isUndefined);
+ *  // <T>(input:T|undefined) => boolean
+ * const isDefined = negate<string | undefined, undefined>(isUndefined);
+ *  // (input: string | undefined) => input is string
+ * ```
+ */
+function negate<T>(predicate: (input: T) => boolean): (input: T) => boolean;
+function negate<T, N>(
+  predicate: (input: T) => input is Include<T, N>,
+): (input: T) => input is Exclude<T, N>;
```

Comment:
I think it's more appropriate to use the `extends` keyword directly instead of using the `Include` type for this case. Additionally, the type signature with a custom type guard for `predicate` should be declared first in order to narrow down the types as intended.

```suggestion
function negate<T, N extends T>(
  predicate: (input: T) => input is N,
): (input: T) => input is Exclude<T, N>;

function negate<T>(predicate: (input: T) => boolean): (input: T) => boolean;
```

Reply:
오버로딩 시그니처의 정의 순서를 바꿔도 제네릭 인자가 하나인 시그니처가 먼저 적용되서 타입가드 형식이 적용되지 않습니다. 그래서 타입 가드 형식의 시그니처만 허용하고, 일반 boolean 리턴형 함수를 타입가드함수로 쉽게 변환하는 방법을 고민중입니다.

Reply:
Please check out this [branch](https://github.com/shine1594/FxTS/tree/feat/util) and review the `throwIf.test.ts` and `negate.test.ts`s files.